azext_iot/models/parsers.py

```diff
@@ -0,0 +1,248 @@
+        self, message: Message, pnp_context, interface_name, origin_device_id
+    ) -> str:
+        message_interface_name = ""
+
+        try:
+            message_interface_name = str(
+                message.annotations.get(INTERFACE_NAME_IDENTIFIER), "utf8"
+            )
+        except Exception:
+            self._errors.append(
+                f"Unable to parse interface_name given a pnp_device. {origin_device_id}. "
+                f"message: {message}"
+            )
+
+        if interface_name != message_interface_name:
+            self._errors.append(
+                f"Inteface name mismatch. {origin_device_id}. "
+                f"Expected: {interface_name}, Actual: {message_interface_name}"
+            )
+
+        return message_interface_name
+
+    def _parse_system_properties(self, message: Message):
+        try:
+            return unicode_binary_map(parse_entity(message.properties, True))
+        except Exception:
+            self._errors.append(
+                f"Failed to parse system_properties for message {message}."
+            )
+            return {}
+
+    def _parse_content_encoding(
+        self, message: Message, system_properties, create_encoding_error
+    ) -> str:
+        content_encoding = ""
+
+        if "content_encoding" in system_properties:
+            content_encoding = system_properties["content_encoding"]
+
+        if not content_encoding:
+            self._errors.append(f"No encoding found for message: {message}")
+            return None
+
+        if create_encoding_error:
+            content_encoding = "Some Random Encoding"
+
+        if "utf-8" not in content_encoding.lower():
+            self._errors.append(
+                f"Unsupported encoding detected: '{content_encoding}'. "
+                f"The currently supported encodings are: {SUPPORTED_ENCODINGS}. "
+                f"System_properties: {system_properties}."
+            )
+            return None
+
+        return content_encoding
+
+    def _parse_content_type(
+        self,
+        content_type_hint,
+        system_properties,
+        origin_device_id,
+        create_custom_header_warning,
+    ) -> str:
+        content_type = ""
+        if content_type_hint:
+            content_type = content_type_hint
+        elif "content_type" in system_properties:
+            content_type = system_properties["content_type"]
+
+        if create_custom_header_warning:
+            content_type = "Some Random Custom Header"
+
+        if not content_type:
+            self._warnings.append(
+                "Content type not found in system_properties. "
+                f"System_properties: {system_properties}"
+            )
+
+        if "application/json" not in content_type.lower():
+            self._warnings.append(
+                "Content type not supported. "
+                f"Content type found: {content_type}. "
+                "Content type expected: application/json. "
```

Comment:
Theoretically regular IoT Hub telemetry does not require content type. Also, general IoT Hub wise application/json is not necessarily required, but you do need application/json if you want to route messages.

Reply:
I've got ideas on how to make this better - i.e. allow callers to specify severity of various errors. Lets leave that for a follow up PR. For now I'll chalk this up to the needs of validate-messages and monitor-events being different

Reply:
#wontfix